src/ATen/native/xpu/sycl/ForeachReduceKernels.cpp

```diff
@@ -194,8 +194,15 @@ std::vector<Tensor> foreach_norm_kernel(
       dtype.has_value() ? dtype.value() : tensors[0].scalar_type();
   const auto options = tensors[0].options();
   auto output_per_tensor_option = options.dtype(toOpMathType(output_dtype));
-  auto res_option = options.dtype(output_dtype);
-  auto ret_per_tensor = at::empty({ntensors}, res_option);
+  std::vector<at::Tensor> ret_per_tensor;
+  ret_per_tensor.reserve(ntensors);
+  const auto res_option = options.dtype(output_dtype);
+  for (int i = 0; i < ntensors; i++) {
+    ret_per_tensor.push_back(at::empty({}, res_option));
+  }
+  sycl::queue q{sycl::property::queue::in_order()};
+  void** tensor_list_addresses = sycl::malloc_shared<void*>((ntensors), q);
```

Comment:
Do not use raw runtime malloc, but helpers provided by PyTorch XPU backend. The usage here, should be,
1. Allocating ping memory
2. Initing meta on ping memory
3. Allocating device memory
4. memcpy from ping to dev